src/System.Management.Automation/engine/runtime/Operations/MiscOps.cs

```diff
@@ -403,31 +403,79 @@ private static string GetParameterText(string parameterName)
 
                 CommandProcessorBase commandProcessor = null;
                 CommandRedirection[] commandRedirection = null;
-                for (int i = 0; i < pipeElements.Length; i++)
-                {
-                    commandRedirection = commandRedirections != null ? commandRedirections[i] : null;
-                    commandProcessor = AddCommand(pipelineProcessor, pipeElements[i], pipeElementAsts[i],
-                                                  commandRedirection, context);
-                }
 
-                var cmdletInfo = commandProcessor?.CommandInfo as CmdletInfo;
-                if (cmdletInfo?.ImplementingType == typeof(OutNullCommand))
-                {
-                    var commandsCount = pipelineProcessor.Commands.Count;
-                    if (commandsCount == 1)
+                var pipelineAst = (PipelineAst)pipeElementAsts[0].Parent;
+                if (pipelineAst.BackgroundProcess)
+                {
+                    // For background jobs rewrite the pipeline as a Start-Job command
+                    var scriptblockBodyString = ((PipelineAst)pipeElementAsts[0].Parent).Extent.Text;
+                    Token[] tokens;
+                    ParseError[] errors;
+                    var pipelineBodyAst = Parser.ParseInput(scriptblockBodyString, out tokens, out errors);
+                    // Prefix variables in the scriptblock with $using: 
+                    var variables = pipelineBodyAst.FindAll(x => x is VariableExpressionAst, true);
+                    System.Text.StringBuilder updatedScriptblock = new System.Text.StringBuilder();
+                    // Make sure that the job runs in the current directory
+                    updatedScriptblock.Append(@"Microsoft.PowerShell.Management\Set-Location $using:pwd; ");
+                    int position = 0;
+                    foreach (var v in variables)
                     {
-                        // Out-Null is the only command, bail without running anything
-                        return;
+                        var vName = ((VariableExpressionAst) v).VariablePath.UserPath;
+                        // Skip variables that don't exist
+                        if (funcContext._executionContext.EngineSessionState.GetVariable(vName) == null)
+                            continue;
+                        // Skip PowerShell magic variables
+                        if (Regex.Match(vName,"^(global:){0,1}(PID$|PS|HOST$|TRUE$|FALSE$|NULL$)", RegexOptions.IgnoreCase|RegexOptions.CultureInvariant).Success == false)
```

Comment:
I don't have a good feeling about not copying **any** variables starting with `PS`. I understand the intent, but I confidently predict it will trip up some folks and their scenario will seem perfectly reasonable.